api/core/v1beta2/machinehealthcheck_types.go

```diff
@@ -60,40 +60,18 @@ type MachineHealthCheckSpec struct {
 	// +required
 	Selector metav1.LabelSelector `json:"selector"`
 
-	// unhealthyNodeConditions contains a list of conditions that determine
-	// whether a node is considered unhealthy. The conditions are combined in a
-	// logical OR, i.e. if any of the conditions is met, the node is unhealthy.
-	//
+	// checks are the checks that are used to evaluate if a Machine is healthy.
 	// +optional
-	// +listType=atomic
-	// +kubebuilder:validation:MinItems=1
-	// +kubebuilder:validation:MaxItems=100
-	UnhealthyNodeConditions []UnhealthyNodeCondition `json:"unhealthyNodeConditions,omitempty"`
+	Checks MachineHealthCheckChecks `json:"checks,omitempty,omitzero"`
 
-	// maxUnhealthy specifies the maximum number of unhealthy machines allowed.
-	// Any further remediation is only allowed if at most "maxUnhealthy" machines selected by
-	// "selector" are not healthy.
-	//
-	// Deprecated: This field is deprecated and is going to be removed in the next apiVersion. Please see https://github.com/kubernetes-sigs/cluster-api/issues/10722 for more details.
-	//
+	// remediation configures if and how remediations are triggered if a Machine is unhealthy.
```

Comment:
can we describe what happens when this is empty? e.g. if empty capi core remediation will take place for machineSets. If the machineSet is owned by a machineDeployment it will the MachineDeployment.Spec.Strategy.Remediation.MaxInFlight...